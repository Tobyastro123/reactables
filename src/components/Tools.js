import React from 'react'
import Button from './Button.js'
import PagesSelector from './PagesSelector.js'
import Search from './Search.js'
import styles from '../css/styles.css'

var CommonConstants = require('./CommonConstants');
var EditorConstants = require('./EditorConstants');
var Lang = require('./Lang');

class Tools extends React.Component {
  getPagesSelection()
  {
    const {
      struct,
      display,
      defaultPerPage,
      perPage,
      updatePerPage,
      perPageRows,
      lang
    } = this.props;

    if (struct.rowsSelector.indexOf(display) === -1) {
      return '';
    }
    return (<PagesSelector
      perPage={perPage}
      updatePerPage={updatePerPage}
      defaultPerPage={defaultPerPage}
      perPageRows={perPageRows}
      lang={lang} />)
  }

  getSearch()
  {
    const {
      struct,
      display,
      search,
      doSearch,
      lang
    } = this.props;

    if (struct.search.indexOf(display) === -1) {
      return '';
    }
    return (<Search
      search={search}
      doSearch={doSearch}
      lang={lang}/>)
  }

  getButtons()
  {
    const {
      selectedRows,
      showPopup,
      defaultPerPage,
      lang,
      tableOpts,
      display
    } = this.props;

    var language = Lang[lang];
    let buttons = [];
    if (typeof tableOpts.buttons !== CommonConstants.UNDEFINED
      && tableOpts.buttonsPosition.indexOf(display) !== -1) {
      tableOpts.buttons.map((btn, i) => {
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_CREATE) {
        buttons[i] = <Button
          active={false}
          action={EditorConstants.ACTION_CREATE}
          showPopup={showPopup}
          key={i}>{language.editor_create}</Button>;
      }
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_EDIT) {
        buttons[i] = <Button
          active={(selectedRows.length === 1) ? false : true}
          selectedRows={selectedRows}
          action={EditorConstants.ACTION_EDIT}
          showPopup={showPopup}
          key={i}>{language.editor_edit}</Button>;
      }
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_REMOVE) {
        buttons[i] = <Button
          active={(selectedRows.length === 0) ? true : false}
          selectedRows={selectedRows}
          action={EditorConstants.ACTION_DELETE}
          showPopup={showPopup}
          key={i}>{language.editor_remove}</Button>;
      }
      });
    }
    return buttons;
  }

  render()
  {
    const {
      search,
      selectedRows,
      showPopup,
      defaultPerPage,
      perPage,
      updatePerPage,
      perPageRows,
      doSearch,
      lang,
      tableOpts
    } = this.props;

    var language = Lang[lang];
    let buttons = [];
    if (typeof tableOpts.buttons !== CommonConstants.UNDEFINED) {
      tableOpts.buttons.map((btn, i) => {
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_CREATE) {
        buttons[i] = <Button
          active={false}
          action={EditorConstants.ACTION_CREATE}
          showPopup={showPopup}
          key={i}>{language.editor_create}</Button>;
      }
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_EDIT) {
        buttons[i] = <Button
          active={(selectedRows.length === 1) ? false : true}
          selectedRows={selectedRows}
          action={EditorConstants.ACTION_EDIT}
          showPopup={showPopup}
          key={i}>{language.editor_edit}</Button>;
      }
      if (btn[EditorConstants.EXTENDED] === EditorConstants.EDITOR_REMOVE) {
        buttons[i] = <Button
          active={(selectedRows.length === 0) ? true : false}
          selectedRows={selectedRows}
          action={EditorConstants.ACTION_DELETE}
          showPopup={showPopup}
          key={i}>{language.editor_remove}</Button>;
      }
      });
    }
    return (
      <div className="gt_head_tools">
        {this.getButtons()}
        {this.getPagesSelection()}
        {this.getSearch()}
        <div className={styles.clear}></div>
      </div>
    )
  }
}

export default Tools
