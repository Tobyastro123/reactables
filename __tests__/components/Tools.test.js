import React from 'react';
import renderer from 'react-test-renderer';
import Tools from '../../src/components/Button.js'

var editor = {
  ajax: 'http://gigatables.loc/editor.php',
  ajaxFiles: 'uploadFiles.php',
  struct: {
    buttons: ['top', 'bottom'] // buttons
  },
  fields: [
    {
      label: "ID",
      name: "id",
      type: 'hidden'
    },
    {// an example of using select - automatically selected if matches with data in table column
      label: "Types:",
      name: "types[]",
      values: [// if select,checkbox,radio etc types - need this pre-set structure of values
        {'key1': 'val1'},
        {'key2': 'val2'}
      ],
      type: 'checkbox', // select,checkbox,radio
//              attrs: [
//                {'multiple':true}
//              ]
    },
    {
      label: "Article title:",
      name: "title",
      type: 'text', // default, other: password, file, select, multiselect etc
      attrs: [
        {'pattern': '^[A-Za-z0-9_]+$'}
      ]
    },
    {
      label: "Description:",
      name: "desc",
      type: 'textarea'
    },
    {
      label: "Date Time:",
      name: "date",
      type: 'date'
    },
    {
      label: "Image:",
      name: "image",
      type: 'file'
    },
  ]
};

it('renders correctly', () => {
  const tree = renderer.create(
    <Tools
    defaultPerPage="100"
    display="bottom"
    lang="en"
    perPage="50"
    perPageRows={[25, 50, 100, 200]}
    search=""
    selectedRows={[]}
    struct={{
      search: ['top', 'bottom'],
      rowsSelector: ['desc', 'top', 'bottom'],
      pagination: ['bottom']
    }}
    tableOpts={{
      buttons: [
        {extended: "editor_create", editor: editor, triggerAfter: (function () {
            console.log('after create');
          }), triggerBefore: (function () {
            console.log('before create');
          })},
        {extended: "editor_edit", editor: editor, triggerBefore: (function () {
          console.log('before edit');
        })},
        {extended: "editor_remove", editor: editor, triggerAfter: (function () {
            console.log('after del');
          })}
      ],
      buttonsPosition: ['top', 'bottom'],
      theme: 'std'
    }}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
