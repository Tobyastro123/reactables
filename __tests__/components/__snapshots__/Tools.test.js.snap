exports[`test renders correctly 1`] = `
<div
  className="gte_buttons_container"
  data-action={undefined}
  onClick={false}>
  <div
    className="gte_button"
    data-action={undefined}>
    <span
      data-action={undefined} />
  </div>
  <div
    className={undefined} />
</div>
`;
